worlds/ahit/__init__.py

```diff
@@ -86,19 +87,28 @@ def generate_early(self):
         if self.is_dw_only():
             return
 
-        # If our starting chapter is 4 and act rando isn't on, force hookshot into inventory
-        # If starting chapter is 3 and painting shuffle is enabled, and act rando isn't, give one free painting unlock
-        start_chapter: ChapterIndex = ChapterIndex(self.options.StartingChapter)
-
-        if start_chapter == ChapterIndex.ALPINE or start_chapter == ChapterIndex.SUBCON:
-            if not self.options.ActRandomizer:
-                if start_chapter == ChapterIndex.ALPINE:
-                    self.multiworld.push_precollected(self.create_item("Hookshot Badge"))
-                    if self.options.UmbrellaLogic:
-                        self.multiworld.push_precollected(self.create_item("Umbrella"))
-
-                if start_chapter == ChapterIndex.SUBCON and self.options.ShuffleSubconPaintings:
+        start_chapter = self.options.StartingChapter
+
+        # Take care of some extremely restrictive starts in other chapters with act shuffle off
+        if start_chapter != ChapterIndex.MAFIA and not self.options.ActRandomizer:
+            if start_chapter == ChapterIndex.ALPINE:
+                self.multiworld.push_precollected(self.create_item("Hookshot Badge"))
+                if self.options.UmbrellaLogic:
+                    self.multiworld.push_precollected(self.create_item("Umbrella"))
+
+                if self.options.ShuffleAlpineZiplines:
+                    ziplines = list(alps_hooks.keys())
```

Comment:
Nitpick: `.keys()` is not necessary here

*Low priority*

Reply:
It only needs the keys here anyways, so I don't see how this is an issue.